snapcraft/internal/meta/snap.py

```diff
@@ -267,6 +279,34 @@ def from_dict(cls, snap_dict: Dict[str, Any]) -> "Snap":
                 for hook_name, hook_dict in snap_dict[key].items():
                     hook = Hook.from_dict(hook_dict=hook_dict, hook_name=hook_name)
                     snap.hooks[hook_name] = hook
+            elif key == "system-usernames":
+                # Process system-username into SystemUsers.
+                raw_usernames = snap_dict["system-usernames"]
+                if not raw_usernames:
+                    continue
+
+                if not isinstance(raw_usernames, dict):
+                    raise RuntimeError(
+                        f"Improperly formatted system-usernames: {raw_usernames}"
+                    )
+
+                for user_name, raw_user in raw_usernames.items():
+                    if isinstance(raw_user, dict):
+                        user = SystemUser.from_dict(
+                            user_dict=raw_user, user_name=user_name
+                        )
+                    elif isinstance(raw_user, str):
+                        try:
+                            scope = SystemUserScope[raw_user.upper()]
+                        except KeyError:
+                            raise errors.SystemUsernamesValidationError(
+                                name=user_name, message=f"scope {raw_user!r} is invalid"
+                            )
+                        user = SystemUser(name=user_name, scope=scope)
+                    else:
```

Comment:
if this is part of passthrough, it probably should be allowed to have other values here.

Reply:
I'm not sure what you mean?

Reply:
no worries, we can cross that bridge later.